src/plugins/discord/sqliteMirror.test.js

```diff
@@ -256,6 +258,37 @@ describe("plugins/discord/sqliteMirror", () => {
       });
     });
 
+    describe("nthMessageIdFromTail", () => {
+      function buildMessages() {
+        const channelId: Snowflake = "1";
+        const authorId: Snowflake = "2";
+
+        const db = new Database(":memory:");
+        const sqliteMirror = new SqliteMirror(db, "0");
+        sqliteMirror.addUser(testUser(authorId));
+        sqliteMirror.addChannel(testChannel(channelId));
+
+        const maxCount = 10;
+        var x = 1;
+        while (x <= maxCount) {
+          const msg = testMessage(String(x), channelId, authorId, x * 2);
+          sqliteMirror.addMessage(msg);
+          x++;
+        }
+        return {sqliteMirror, channelId};
+      }
+      it("retrieves nth message from tail", () => {
+        const {sqliteMirror, channelId} = buildMessages();
+        expect(sqliteMirror.nthMessageIdFromTail(channelId, 3)).toBe("7");
+      });
+      it("handles n greater than total message count", () => {
+        const {sqliteMirror, channelId} = buildMessages();
+        expect(sqliteMirror.nthMessageIdFromTail(channelId, 20)).toBe(
+          undefined
```

Comment:
I kind of think the caller is in a better position to handle the case where n > total message count, rather than `sqliteMirror`

Reply:
I agree, for the caching strategy we don't need any extra information than the return type being nullable. If `n` > total messages, it simply means we should (re)load all messages in that channel :]